src/core/olaf.ts

```diff
@@ -0,0 +1,70 @@
+import * as TelegramBot from 'node-telegram-bot-api';
+import { Message } from 'node-telegram-bot-api';
+
+import preferencesDashboard from '../dashboard/preferences-dashboard';
+import MessageSender from './messageSender';
+import MessageHandler from './messageHandler';
+import MessageRouter from './messageRouter';
+import EndUseCaseResponse from '../classes/EndUseCaseResponse';
+import UseCase from '../interfaces/useCase';
+
+export default class Olaf {
+  private telegramBot;
+
+  private dashboard;
+
+  private messageHandler;
+
+  private messageRouter;
+
+  private messageSender;
+
+  private activeUseCase: UseCase;
+
+  constructor() {
+    this.telegramBot = new TelegramBot(process.env.BOT_TOKEN, { polling: true });
+    this.dashboard = preferencesDashboard;
+    this.messageHandler = new MessageHandler(this.telegramBot);
+    this.messageRouter = new MessageRouter();
+    this.messageSender = new MessageSender(this.telegramBot);
+    this.activeUseCase = null;
+
+    // TODO register all use cases here
+    // this.messageRouter.registerUseCase(new XUseCase())
+  }
+
+  start() {
+    this.dashboard.listen(process.env.PORT, () => {
+      console.log(`Dashboard is running in http://localhost:${process.env.PORT}`);
+    });
+
+    this.telegramBot.on('message', (msg) => this.handleTelegramMessage(msg));
+  }
+
+  private async handleTelegramMessage(originalMessage: Message): Promise<void> {
+    this.messageSender.setChatId(originalMessage.chat.id);
+
+    try {
+      // Extract message, including speech recognition
+      const message = await this.messageHandler.extractMessage(originalMessage);
+      // Find matching use case
+      if (!this.activeUseCase) {
+        this.activeUseCase = await this.messageRouter.findUseCase(message);
+      }
+      if (!this.activeUseCase) {
+        throw new Error('Invalid use case');
+      }
```

Comment:
how bout else or elseif

Reply:
Ich hab das Handling von fehlenden Use Cases in den MessageRouter verschoben, somit ist hier nur noch ein if